test/unit/lib/user_parameter_sanitiser_test.rb

```diff
@@ -0,0 +1,84 @@
+require 'test_helper'
+
+class UserParameterSanitiserTest < ActiveSupport::TestCase
+  context "with Plain Old Ruby Hash for params" do
+    setup do
+      permitted_params_by_role = { normal: [:name, :email] }
+      user_params = { name: "Anne", email: "anne@anne.com" }
+
+      @sanitised_params = UserParameterSanitiser.new(
+        user_params: user_params,
+        current_user_role: :normal,
+        permitted_params_by_role: permitted_params_by_role,
+      ).sanitise
+    end
+
+    should "return permitted params object" do
+      assert_instance_of ActionController::Parameters, @sanitised_params
+      assert @sanitised_params.permitted?
+    end
+  end
+
+  context "with ActionController params object" do
+    setup do
+      permitted_params_by_role = { normal: [:name, :email] }
+      user_params = ActionController::Parameters.new(
+        { name: "Bill", email: "bill@bill.com" }
+      )
+
+      @sanitised_params = UserParameterSanitiser.new(
+        user_params: user_params,
+        current_user_role: :normal,
+        permitted_params_by_role: permitted_params_by_role,
+      ).sanitise
+    end
+
+    should "return permitted params object" do
+      assert_instance_of ActionController::Parameters, @sanitised_params
+      assert @sanitised_params.permitted?
```

Comment:
Worth checking the permitted params haven't been lost? I know it's a thin wrapper around `ActionController::Parameters` but might be worth checking